lib/improver/nbhood/recursive_filter.py

```diff
@@ -73,29 +79,35 @@ def __init__(self, alpha_x=None, alpha_y=None, iterations=None,
                 originally masked.
 
         Raises:
-            ValueError: If alpha_x is not set such that 0 < alpha_x < 1
-            ValueError: If alpha_y is not set such that 0 < alpha_y < 1
+            ValueError: If alpha_x is not set such that 0 < alpha_x <= 0.5
+            ValueError: If alpha_y is not set such that 0 < alpha_y <= 0.5
             ValueError: If number of iterations is not None and is set such
-                        that iterations is not >= 1
+                        that iterations is less than 1.
 
-        """
-        if alpha_x is not None:
-            if not 0 < alpha_x < 1:
-                raise ValueError(
-                    "Invalid alpha_x: must be > 0 and < 1: {}".format(
-                        alpha_x))
+        Warns:
+            UserWarning:
+                If iterations is higher than 2.
 
-        if alpha_y is not None:
-            if not 0 < alpha_y < 1:
-                raise ValueError(
-                    "Invalid alpha_y: must be > 0 and < 1: {}".format(
-                        alpha_y))
 
+        """
+        alpha_error = ("alpha must be less than 0.5. A large alpha value"
+                       "leads to poor conservation of probabilities: ")
+        if alpha_x is not None and not 0 < alpha_x <= 0.5:
+            message = alpha_error if alpha_x > 0.5 else ''
+            message += "Invalid alpha_x: must be > 0 and <= 0.5: {}"
+            raise ValueError(message.format(alpha_x))
+        if alpha_y is not None and not 0 < alpha_y <= 0.5:
+            message = alpha_error if alpha_y > 0.5 else ''
+            message += "Invalid alpha_y: must be > 0 and <= 0.5: {}"
+            raise ValueError(message.format(alpha_y))
         if iterations is not None:
-            if not iterations >= 1:
+            if iterations < 1:
                 raise ValueError(
                     "Invalid number of iterations: must be >= 1: {}".format(
                         iterations))
+            if iterations > 2:
+                warnings.warn(
+                    "Iterations of two will lead to poorer conservation.")
```

Comment:
`iterations > 2`, yet the message implies `iterations == 2` - additionally, is this message actually useful to an ed user, or should it be obvious from the algorithm? What if `iterations == 3`, is it poor then also?

Reply:
The ticket asks for a warning if more than 2 iterations are used, so it is the message that needs updating here. How about "More than two iterations degrades the conservation of probability assumption."

Reply:
Thank you. done.